pallets/gear/src/manager/task.rs

```diff
@@ -85,15 +89,156 @@ pub fn get_maximum_task_gas<T: Config>(task: &ScheduledTask<T::AccountId>) -> Ga
     }
 }
 
+fn clean<T: Config>(program_id: ProgramId, gas_reservation_map: GasReservationMap)
+where
+    T::AccountId: Origin,
+{
+    // clean wait list from the messages
+    let reason = MessageWokenSystemReason::ProgramGotInitialized.into_reason();
+    WaitlistOf::<T>::drain_key(program_id).for_each(|entry| {
+        let message = Pallet::<T>::wake_dispatch_requirements(entry, reason.clone());
+
+        QueueOf::<T>::queue(message)
+            .unwrap_or_else(|e| unreachable!("Message queue corrupted! {:?}", e));
+    });
+
+    ExtManager::<T>::remove_gas_reservation_map(program_id, gas_reservation_map);
+}
+
+fn handle_paused<T: Config>(
+    program_id: ProgramId,
+    gas_reservation_map: GasReservationMap,
+    page_count: u32,
+) -> Gas
+where
+    T::AccountId: Origin,
+{
+    clean::<T>(program_id, gas_reservation_map);
+
+    Pallet::<T>::deposit_event(Event::ProgramChanged {
+        id: program_id,
+        change: ProgramChangeKind::Paused,
+    });
+
+    let gas = <T as Config>::WeightInfo::tasks_pause_program(page_count).ref_time();
+    log::trace!("Task gas: tasks_pause_program = {gas}");
+
+    gas
+}
+
+fn handle_uninitialized<T: Config>(program_id: ProgramId, init_message_id: MessageId) -> Gas
```

Comment:
As soon as we have special directory for `manager` you can move all `handle_` functions  to separate file pause.rs